test/scripts/jenkins_setup_parallel_workspace.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+set -e
+
+CURRENT_DIR=$(pwd)
+
+# Copy everything except node_modules into the current workspace
+rsync -a ${WORKSPACE}/kibana/* . --exclude node_modules
+rsync -a ${WORKSPACE}/kibana/.??* .
+
+# Symlink all non-root, non-fixture node_modules into our new workspace
+cd ${WORKSPACE}/kibana
+find . -type d -name node_modules -not -path '*__fixtures__*' -not -path './node_modules*' -prune -print0 | xargs -0I % ln -s "${WORKSPACE}/kibana/%" "${CURRENT_DIR}/%"
+find . -type d -wholename '*__fixtures__*node_modules' -not -path './node_modules*' -prune -print0 | xargs -0I % cp -R "${WORKSPACE}/kibana/%" "${CURRENT_DIR}/%"
+cd "${CURRENT_DIR}"
+
+# Symlink all of the individual root-level node_modules into the node_modules/ directory
+mkdir -p node_modules
+ln -s ${WORKSPACE}/kibana/node_modules/* node_modules/
+ln -s ${WORKSPACE}/kibana/node_modules/.??* node_modules/
+
+# Copy a few node_modules instead of symlinking them. They don't work correctly if symlinked
+unlink node_modules/@kbn
+unlink node_modules/css-loader
+unlink node_modules/style-loader
+
+# packages/kbn-optimizer/src/integration_tests/basic_optimization.test.ts will fail if this is a symlink
+unlink node_modules/val-loader
```

Comment:
I took a look at this, and I think I misunderstood what you meant. It's probably the dynamic import names that webpack generates that were breaking this tests? Like `_node_modules_val_loader_dist_cjs_js_key_bar_kbn_ui_shared_deps_public_path_module_creator_js__WEBPACK_IMPORTED_MODULE_0___default`

Reply:
Yep, they ended up as `_kibana_node_modules_val_loader_dist_cjs_js_key_bar_kbn_ui_shared_deps_public_path_module_creator_js__WEBPACK_IMPORTED_MODULE_0___default` (note the `_kibana` prefix)